pkg/pgmodel/sql_reader.go

```diff
@@ -357,21 +364,24 @@ func (q *pgxQuerier) querySingleMetric(metric string, filter metricTimeRangeFilt
 	}
 
 	// TODO this allocation assumes we usually have 1 row, if not, refactor
-	tsRows := appendTsRows(make([]timescaleRow, 0, 1), rows)
-	return tsRows, topNode, nil
+	tsRows, err := appendTsRows(make([]timescaleRow, 0, 1), rows)
+	return tsRows, topNode, err
 }
 
-func appendTsRows(out []timescaleRow, in pgx.Rows) []timescaleRow {
+func appendTsRows(out []timescaleRow, in pgx.Rows) ([]timescaleRow, error) {
+	if in.Err() != nil {
```

Comment:
I think this should be at the end of the function after the in.Next calls, not the beginning

Reply:
I think we need to check in the beginning also, in case it comes in with an error